### PR TITLE
Maps playground

### PR DESCRIPTION
**Pull request by yainporter**
## Type of change
- [x] New feature
- [ ] Bug Fix
- [x] Test
- [ ] Refactor/Style

## Description
Relevant User Story(s):
*Brief summary of the change(s) included in this pull request*

- CSS styling
- Google Maps API
- Search PORO
- Discover Facade
- Geocoder Gem
- Javascript stuff
- Removed all @users, and replaced with current_user
- Removed `resources :users` path, no longer needed with Devise
- Added rescue for `TripService.create_trip` in case of 400 error from API

This description closes #

## Check the correct boxes
- [ ] This broke nothing
- [X] This broke some stuff
- [ ] This broke everything

## Testing Changes
- [ ] No Tests have been changed
- [X] Some Tests have been changed
- [ ] All of the Tests have been changed

- I changed the buttons to link, and styled them to look like buttons
- Pulling in Draper data messed up with the Date formatting

*Please describe what in the world happened if tests are broken*

## Checklist:
- [ ] My code has no unused/commented out code
- [x] I have reviewed my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have fully tested my code

### CODE REVIEW

**Review by Selena730 — APPROVED** [review]
Thank you Yain and Igor 🌟
Review was done live

File: .gitignore [hunk 1]
```diff
@@ -37,3 +37,10 @@
 .env
 /app/assets/builds/*
 !/app/assets/builds/.keep
```
Comment (IgorrMagalhaess) [inline .gitignore:39]: I think we can add vendor/assets to git ignore too. Or will this break the application?

File: Gemfile [hunk 2]
```diff
@@ -39,6 +39,14 @@ gem 'omniauth-google-oauth2'
 gem "omniauth-rails_csrf_protection"
 
 gem 'dotenv-rails'
+
+# makick gem
+gem "mapkick-rb"
```
Comment (IgorrMagalhaess) [inline Gemfile:44]: We are not using this anymore, note to remove later

File: app/javascript/application.js [hunk 3]
```diff
@@ -1,3 +1,126 @@
...
+      position: latLng,
+      map: map,
+    });
+    map.panTo(latLng);
+
+    map.setZoom(15)
+  }
+
+  // Used to wait for the #map to appear, if it does, is it a String?
+  // If it is a String, then return a promise
```
Comment (IgorrMagalhaess) [inline app/javascript/application.js:88]: The comments here are super helpful, sorry I didn't do a better job when I started this

File: config/routes.rb [hunk 4]
```diff
@@ -15,6 +15,7 @@
   get '/auth/google_oauth2/callback', to: 'sessions#google', as: 'auth_google'
 
   resources :users, only: [:show, :new, :create] do
```
Comment (IgorrMagalhaess) [inline config/routes.rb:17]: We need to figure out how to adapt this since we will not use users controller anymore.

File: app/views/users/show.html.erb
Comment (Selena730) [inline app/views/users/show.html.erb:1]: Thank you!

**Review by Selena730 — APPROVED** [review]

**Review by hoadam — APPROVED** [review]

**Pull request merged**